### PR TITLE
Adds README for backend

### PR DESCRIPTION
**Pull request by alyshahudson**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+### Kotlin
+
+* The backend is written in Kotlin. Make sure you have a compatible IDE such as [IntelliJ](https://www.jetbrains.com/idea/download/index.html?_ga=2.137859766.761208892.1584829709-1795868819.1584829709#section=mac).
+
+
+### Java 11
+
+* This project requires Java version 11 which can be dowloaded [here](https://www.oracle.com/java/technologies/javase-jdk11-downloads.html)
+
+### Maven
```
Comment (toadzky) [inline README.md]: these are best installed with a package manager, so we probably want a comment about installing that, with `brew` for mac

File: README.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+* In a different terminal window, first port is http endpoint, second is bolt port :7687
+
+```sh
+$ docker run -d -p7474:7474 -p7687:7687 --name=zerboase-db neo4j:latest
+```
+
+* Open http://localhost:7474/ in your browser
+The default username and password for Neo4j is 
+username: Neo4j 
+Password: Neo4j
```
Comment (toadzky) [inline README.md]: should be all lowercase for both

File: README.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+You will then be asked to set a new password, make this password zerobase is currently used “zerobase”
```
Comment (toadzky) [inline README.md]: This sentence needs fixing. I think it's trying to say "set the password to 'zerobase'" but there's a part after that I'm not sure what it's trying to say.

File: README.md [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+### Project structure
+
+
```
Comment (toadzky) [inline README.md]: is there something that is supposed to be in this section?

File: README.md [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+### Project
+
+* After cloning the project there are many ways to deploy it locally. The following directions pertain to deployment via the IntelliJ IDE. In whichever environment you choose, ensure you're using Java 11 and set the environment variable `GRAPHENEDB_PASSWORD=zerobase`. Run the server via `server src/main/resources/config.yml`
+
+
+*In IntelliJ
+	* Navigate to File/Project Structure. Update the JDK home path to Java 11.
+	* In the file path src/main/kotlin/models/ open `Main.kt`
+	* In `Main.kt` right click on the run button next to main and click Edit Run Configuration
+	* ![main](https://github.com/alh2202/smart-tracing-api/blob/master/main.png)
```
Comment (toadzky) [inline README.md]: the full url for images shouldn't point to your repo. this should probably be a relative url. also, i don't see the image files added to the repo as part of the PR. same for the one below

**Review by toadzky — CHANGES_REQUESTED** [review]

File: README.md [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+
+# TODO
+
+*
+
+## Set up for the project
+
+### Kotlin
+
+* The backend is written in Kotlin. Make sure you have a compatible IDE such as [IntelliJ](https://www.jetbrains.com/idea/download/index.html?_ga=2.137859766.761208892.1584829709-1795868819.1584829709#section=mac).
```
Comment (toadzky) [inline README.md]: this link has the mac section in it. trim that off just to make it easier for people who aren't on mac

**Comment by mbuchel** [comment]
We use Kotlin version 1.3.70, OpenJDK version 9, any version of maven should work.

Overall the rest i feel is decent enough to teach people how to start developing. Below are just some more observations I had.

* It is not normally a good idea to suggest the password to use, this is the exact reason we use environment variables.
* There is also some conflicting information, for example in the first Project section you state: Run the server via `server src/main/resources/config.yml`.
* Unfortunately, here I get server is not found, so you are likely missing a configuration somewhere there.
* The "run this to build" command is the command you run to make the server run, and does not build the server, `mvn install` is necessary to rebuild the program.
* Why are you suggesting to do a `mvn clean install` when all you need is a `mvn install` when you compile for the first time?

Here are some issues I ran into running/installing the project on my test setups (NixOS, GuixSD, Arch, FreeBSD):

* Arch - MVN was complaining: "Rule 1: org.apache.maven.plugins.enforcer.RequireJavaVersion failed with the message: Detected JDK Version: 1.8.0-242 is not in the allowed range 9" (after installing java-11-openjdk)
   * Solution was to run `sudo archlinux-java set java-11-openjdk`
* NixOS - No issues on my setup, can create an environment shell script if we have multiple users of this wonderful os.
* GuixSD - No issues on my setup, but can create an environment script to install for testing if necessary, and if we have enough people using this.
* FreeBSD - Highly suggest against this, as in the future we may not be able to support it based on how many more dependencies we add to it, works for now, just recall to switch the openjdk path to use the correct version.

**Comment by toadzky** [comment]
@mbuchel we are using java11 because it's the current LTS version. as for which implementation, that's a personal preference not something dictated by us. they can use oracle, openjdk, or adoptopenjdk - doesn't matter.

also, we aren't going to include instructions for how to install or configure java on every variation of linux or other OS distros. if someone wants to use linux or bsd, that's fine, but it's up to them to configure their environment correctly.

**Review by toadzky — APPROVED** [review]

**Pull request merged**